### ISSUE TITLE
Java (Tomcat) app that runs in Google cloud doesnt' connect externally

### ISSUE THREAD

**Issue by indykish** [cloudius-systems/osv#254]
### 1.  An app can't be accessed externally after starting it in Google.

I have all the firewall rules enabled. 

The serialport output is here. 

```
Booting from 0000:7c00
OSv v0.06-rc2-114-g82b69aa
------------------------------------------------------------
Server listening on TCP port 5001
TCP window size: 64.0 KByte (default)
------------------------------------------------------------
03-Apr-2014 09:22:06.733 INFO [Thread-4] org.apache.catalina.core.AprLifecycleListener.init The APR based Apache Tomcat Native library which allows optimal performance in production environments was not found on the java.library.path: /usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
03-Apr-2014 09:22:07.911 INFO [Thread-4] org.apache.coyote.AbstractProtocol.init Initializing ProtocolHandler ["http-nio-8081"]
03-Apr-2014 09:22:08.017 INFO [Thread-4] org.apache.tomcat.util.net.NioSelectorPool.getSharedSelector Using a shared selector for servlet write/read
03-Apr-2014 09:22:08.023 INFO [Thread-4] org.apache.coyote.AbstractProtocol.init Initializing ProtocolHandler ["ajp-nio-8009"]
03-Apr-2014 09:22:08.030 INFO [Thread-4] org.apache.tomcat.util.net.NioSelectorPool.getSharedSelector Using a shared selector for servlet write/read
03-Apr-2014 09:22:08.033 INFO [Thread-4] org.apache.catalina.startup.Catalina.load Initialization processed in 3065 ms
03-Apr-2014 09:22:08.235 INFO [Thread-4] org.apache.catalina.core.StandardService.startInternal Starting service Catalina
03-Apr-2014 09:22:08.237 INFO [Thread-4] org.apache.catalina.core.StandardEngine.startInternal Starting Servlet Engine: Apache Tomcat/8.0.5
03-Apr-2014 09:22:08.309 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory /usr/tomcat/webapps/docs
03-Apr-2014 09:22:14.273 INFO [localhost-startStop-1] org.apache.catalina.util.SessionIdGenerator.createSecureRandom Creation of SecureRandom instance for session ID generation using [SHA1PRNG] took [5,023] milliseconds.
03-Apr-2014 09:22:14.351 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory /usr/tomcat/webapps/examples
03-Apr-2014 09:22:17.012 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory /usr/tomcat/webapps/manager
03-Apr-2014 09:22:17.165 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory /usr/tomcat/webapps/ROOT
03-Apr-2014 09:22:17.262 INFO [localhost-startStop-1] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory /usr/tomcat/webapps/host-manager
03-Apr-2014 09:22:17.504 INFO [Thread-4] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["http-nio-8081"]
03-Apr-2014 09:22:17.511 INFO [Thread-4] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["ajp-nio-8009"]
03-Apr-2014 09:22:17.514 INFO [Thread-4] org.apache.catalina.startup.Catalina.start Server startup in 9479 ms
```

The network doesn't get attached when it runs on google and hence I am unable to get to http://public_ip:8081 or 8000.  

The same image works just fine locally which means I can manage using port 8000, and 8081 but when I upload my own war file, it crashes. I had to reboot my image.
### 2. Also the osv image size is big 200MB with today's latest code commit but Ubuntu saucy 236MB.

I can install java internally later on in ubuntu and play with it.  Couldn't see the metrics in OSv as well, probably newrelic might be the way to go, 
#### code(0.6rc2) seems to be on the beta level.

I can start an Ubuntu image faster on the same cloud today. 
Why would I have to choose ubuntu to OSv ? What may be the business drivers for choosing  OSv to Ubuntu or Redhat ?

**Comment by VivekRoshan** [comment]
Hi, did you resolve the issue? I have a same problem, where it shoes server startup in xxxx seconds but doesn't show up in its externalip:port. Please help me.

**Comment by nyh** [comment]
Yes, this issue was solved 3 years ago, in commit f43ffa16c5e547a2fe27cac6360f850a6d278d8a - by fixing bugs in the DHCP protocol parsing.

However, since the DHCP code *has* changed in the time that passed, it is possible that a new bug was introduced. Or this might just be a configuration error on your side (e.g., you don't have a DHCP server running properly). Please open a new issue about this and describe your problem with more details. If possible run OSv with verbose diagnostics (e.g., scripts/run.py -V) and see if there are any DHCP-related messages.
